Scripts/Biochemistry/Add_New_Reactions.py

```diff
@@ -0,0 +1,225 @@
+               "code":"null","stoichiometry":"null","equation":"null","definition":"null",
+               "reversibility":"=","direction":"=","deltag":"10000000","deltagerr":"10000000",
+               "status":"NB","is_obsolete":0,"is_transport":0,
+               "abstract_reaction":"null","pathways":"null","ec_numbers":"null",
+               "compound_ids":"null","linked_reaction":"null","notes":"null","source":""}
+
+Compounds_Alias_Dict=CompoundsHelper.loadMSAliases()
+Source_Alias_Dict = dict()
+for msid in Compounds_Alias_Dict.keys():
+    for source in Compounds_Alias_Dict[msid].keys():
+        if(source not in Source_Alias_Dict):
+            Source_Alias_Dict[source]=dict()
+        for alias in Compounds_Alias_Dict[msid][source]:
+            if(alias not in Source_Alias_Dict[source]):
+                Source_Alias_Dict[source][alias]=list()
+            Source_Alias_Dict[source][alias].append(msid)
+
+Original_Alias_Dict=ReactionsHelper.loadMSAliases()
+New_Alias_Count=dict()
+
+Names_Dict = ReactionsHelper.loadNames()
+All_Names = dict()
+New_Name_Count=dict()
+for msid in sorted(Names_Dict.keys()):
+    for name in Names_Dict[msid]:
+        All_Names[name]=1
+
+ECs_Dict = ReactionsHelper.loadECs()
+All_ECs = dict()
+New_EC_Count=dict()
+for msid in sorted(ECs_Dict.keys()):
+    for name in ECs_Dict[msid]:
+        All_ECs[name]=1
+
+#Find last identifier and increment
+last_identifier = list(sorted(Reactions_Dict.keys()))[-1]
+identifier_count = int(re.sub('^rxn','',last_identifier))
+
+#If a reaction, after removing cpd redundancy, is empty
+#We use a placeholder
+Empty_Rxn_ID="rxn14003"
+
+New_Rxn_Count=dict()
+Headers=list()
+rxns=list()
+missing_cpds=dict()
+with open(Biochem_Root+Biochem+"_Reactions.tbl") as fh:
+    for line in fh.readlines():
+        line=line.strip()
+        if(len(Headers)==0):
+            Headers=line.split('\t')
+            continue
+
+        rxn=dict()
+        array=line.split('\t',len(Headers))
+        for i in range(len(Headers)):
+            rxn[Headers[i]]=array[i]
+
+        #Retrieve identifiers from within equation
+        #Split based on whitespace, and remove compartment index
+        original_cpd_array=rxn['EQUATION'].split(' ')
+        new_cpd_array=list()
+        for i in range(len(original_cpd_array)):
+            if(re.search('\[[01]\]$',original_cpd_array[i])):
+                new_cpd_array.append(re.sub('\[[01]\]$','',original_cpd_array[i]))
+
+        all_matched=1
+        for new_cpd in new_cpd_array:
+            if(new_cpd in Source_Alias_Dict[Biochem]):
+                msid = sorted(Source_Alias_Dict[Biochem][new_cpd])[0]
+                rxn['EQUATION'] = re.sub(new_cpd,msid,rxn['EQUATION'])
+            else:
+                missing_cpds[new_cpd]=1
+                all_matched=0
+
+        if(all_matched==0):
+            print("Warning: missing "+Biochem+" identifiers for reaction "+rxn['ID']+": "+rxn['EQUATION'])
```

Comment:
F-strings provide an better way to do this in python 3: https://realpython.com/python-f-strings/
```suggestion
            print(f"Warning: missing {Biochem} identifiers for reaction {rxn['ID']}: {rxn['EQUATION']}")
```

Reply:
But this is low priority. concat is still legible